File: frontend/projects/admin/src/app/dto/stock-reservation.dto.ts
/** Cycle de vie d'une réservation de stock. */
export type ReservationStatut = 'ACTIVE' | 'RELEASED' | 'CONSUMED' | 'EXPIRED';

export interface StockReservationDTO {
  id: number;
  offerId: number;
  commandeId?: number;
  commandeReference?: string;
  quantite: number;
  statut: ReservationStatut;
  /** Échéance calculée par le serveur : ne pas recalculer côté client. */
  expiresAt?: string;
  createdAt?: string;
  closedAt?: string;
  closeReason?: string;
}

File: backend/src/main/java/com/vollailelink/backend/dto/StockReservationDTO.java
package com.vollailelink.backend.dto;

import com.vollailelink.backend.model.enums.ReservationStatut;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StockReservationDTO {
    private Long id;
    private Long offerId;
    private Long commandeId;
    private String commandeReference;
    private Integer quantite;
    private ReservationStatut statut;
    private OffsetDateTime expiresAt;
    private OffsetDateTime createdAt;
    private OffsetDateTime closedAt;
    private String closeReason;
}
